power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -328,13 +329,18 @@ inline auto get_edge_weights(TransformerGraph const& graph) -> TrafoGraphEdgePro
         // side via the bidirectional edge (if it exists). For delta configuration ABC, the above
         // situations can happen.
         // The logic still holds in meshed grids, albeit operating a more complex graph.
-        if (edge_src_rank != edge_tgt_rank - 1) {
-            throw AutomaticTapInputError("The control side of a transformer regulator should be relatively further "
-                                         "away from the source than the tap side.\n");
+        if (is_unreachable(edge_res)) {
+            continue;
         }
```

Comment:
this makes the control flow more difficult to understand. i prefer the nested if-statements in this case

Reply:
Yes, I remember you mentioned this in the original PR of is_unreachable too :D